webui/react/src/components/kit/LineChart.tsx

```diff
@@ -72,39 +94,62 @@ export const LineChart: React.FC<Props> = ({
         { label: xLabel || 'X' },
         ...series.slice(1).map((serie, idx) => {
           return {
-            label: serie.name ?? `Series ${idx + 1}`,
+            label: metric.name ?? `Series ${idx + 1}`,
+            points: { show: false },
             scale: 'y',
             spanGaps: true,
-            stroke: serie.color ?? glasbeyColor(idx),
+            stroke: seriesColors[idx],
+            type: 'line',
+            width: 2,
           };
         }),
       ],
     };
-  }, [series, height, scale, showLegend, xLabel, yLabel]);
+  }, [series, seriesColors, height, metric.name, scale, xLabel, yLabel]);
 
   return (
     <>
-      {title && <h5 className={css.chartTitle}>{title}</h5>}
+      {metric.name && <h5 className={css.chartTitle}>{metric.name}</h5>}
       <UPlotChart
         data={series.map((s) => s.data) as AlignedData}
         focusIndex={focusedSeries}
         options={chartOptions}
       />
+      {showLegend && (
+        <div className={css.legendContainer}>
+          {series.slice(1).map((s, idx) => (
+            <li className={css.legendItem} key={idx}>
+              <span className={css.colorButton} style={{ color: seriesColors[idx] }}>
+                &mdash;
+              </span>
+              {s.metricType}_{metric.name}
+            </li>
+          ))}
+        </div>
+      )}
     </>
   );
 };
 
 interface GroupProps {
+  chartSettingsControls?: boolean;
```

Comment:
Controls should always be visible but you probably want to automatically hide the axis control if only one axis is passed.

Reply:
✅